In this study, our primary goal is to achieve a chromosome-level assembly and annotation of distinct accessions of Arabidopsis thaliana, specifically focusing on the Shahdara (Sha) accession originating from Tajikistan. The comparative analysis will extend to include other accessions from other teams such as An-1 (Antwerp, Belgium), C24 (Coimbra, Portugal), Cvi-0 (Cape Verde Islands), and Ler-3 (Gorz√≥w Wielkopolski, Poland). The goal is to unravel the genomic intricacies of Arabidopsis thaliana by employing cutting-edge sequencing technologies and assembly strategies. The assembly process involves the integration of data from Flye, Canu, and Trinity assemblers, followed by polishing using Pilon to enhance the accuracy of the assemblies. Through this, we aim to produce high-quality, chromosome-level assemblies that provide a comprehensive view of the genomic landscape. Beyond the technical aspects, the study aims to delve into the evolutionary dynamics of these accessions. By comparing the assembled genomes, we seek to uncover patterns of genome diversity, structural variations, and potential adaptive changes that have evolved through both spatial and temporal dimensions. Each accession, with its unique geographical origin, presents an opportunity to explore how environmental factors may have shaped the genomic architecture of Arabidopsis thaliana populations across different regions.
